x-pack/plugins/endpoint/public/applications/endpoint/store/endpoint_list/saga.ts

```diff
@@ -6,17 +6,29 @@
 
 import { CoreStart } from 'kibana/public';
 import { SagaContext } from '../../lib';
-import { EndpointListAction } from './action';
+import { AppAction } from '../action';
+import { endpointListPageIndex, endpointListPageSize } from './selectors';
 
 export const endpointListSaga = async (
-  { actionsAndState, dispatch }: SagaContext<EndpointListAction>,
+  { actionsAndState, dispatch }: SagaContext<AppAction>,
   coreStart: CoreStart
 ) => {
   const { post: httpPost } = coreStart.http;
 
-  for await (const { action } of actionsAndState()) {
-    if (action.type === 'userEnteredEndpointListPage') {
-      const response = await httpPost('/api/endpoint/endpoints');
+  for await (const { action, state } of actionsAndState()) {
+    if (
+      (action.type === 'userNavigatedToPage' && action.payload === 'managementPage') ||
+      action.type === 'userPaginatedEndpointListTable'
+    ) {
+      const pageIndex = endpointListPageIndex(state.endpointList);
+      const pageSize = endpointListPageSize(state.endpointList);
+      const response = await httpPost('/api/endpoint/endpoints', {
```

Comment:
probably ok for now, but when we move this to Middleware, we should also handle errors (or `throw`'s) here. Question will be: what should we do with the errors from the UI standpoint.